File: Haskell/734A.hs
import Data.Char (isSpace, isAlpha)

main :: IO ()
main = interact $ solve . dropWhile isSpace . reverse . dropWhile (not . isAlpha)

solve :: String -> String
solve x = case compare anton danik of
  LT -> "Danik"
  GT -> "Anton"
  EQ -> "Friendship"
  where
    anton = length $ filter (== 'A') x
    danik = length x - anton;

File: CPP/734A.cpp
#include <iostream>

using namespace std;

int anton, danik;

int main () {
  int n;  
  cin >> n;

  char w;
  while(cin >> w) 
  {
    if (w == 'A') anton++;
    else          danik++;
  }

  if (anton > danik)      cout << "Anton" << endl;
  else if (anton < danik) cout << "Danik" << endl;
  else                    cout << "Friendship" << endl;
}